crypto/rsa/rsa_pmeth.c

```diff
@@ -653,9 +653,8 @@ static int pkey_rsa_ctrl_str(EVP_PKEY_CTX *ctx,
         BIGNUM *pubexp = NULL;
         if (!BN_asc2bn(&pubexp, value))
             return 0;
-        ret = EVP_PKEY_CTX_set_rsa_keygen_pubexp(ctx, pubexp);
-        if (ret <= 0)
-            BN_free(pubexp);
+        ret = EVP_PKEY_CTX_set1_rsa_keygen_pubexp(ctx, pubexp);
+        BN_free(pubexp);
```

Comment:
I suspect you might be breaking something else with these changes..
pkey_rsa_ctrl also has this..
```
    case EVP_PKEY_CTRL_RSA_KEYGEN_PUBEXP:
        if (p2 == NULL || !BN_is_odd((BIGNUM *)p2) || BN_is_one((BIGNUM *)p2)) {
            RSAerr(RSA_F_PKEY_RSA_CTRL, RSA_R_BAD_E_VALUE);
            return -2;
        }
        BN_free(rctx->pub_exp);
        rctx->pub_exp = p2;
        return 1;
```

Reply:
Why? The original code frees the local BIGNUM only in the event an error is returned, which makes sense because in the success case the set0-style API took over custody. Because I changed to set1-style, where a copy is always made, the local BIGNUM should always be freed afterward. I don't see how the code you've cited has anything to do with it.

Reply:
Was just making sure you were aware of that path..
The free still does a BN_free(rctx->pub_exp).. so it should be ok.